src/py/flwr/simulation/run_simulation.py

```diff
@@ -201,8 +201,11 @@ def _main_loop(
     f_stop = asyncio.Event()
     serverapp_th = None
     try:
+        # Create run (with empty fab_id and fab_version)
+        run_id = state_factory.state().create_run("", "")
+
         # Initialize Driver
-        driver = InMemoryDriver(state_factory)
+        driver = InMemoryDriver(run_id=run_id, state_factory=state_factory)
 
         if run_id:
             _init_run_id(driver, state_factory, run_id)
```

Comment:
The idea behind having this logic was so the `Simulation Engine` uses a `run_id` passed from an external component (e.g. the `SuperExec`). With the changes done a few lines above, this behaviour is changed (since `run_id` passed will be overridden by what `create_run()` returns.

Reply:
Would this be enough to retain the functionality?
```
    try:
        # Create run (with empty fab_id and fab_version)
        run_id_ = state_factory.state().create_run("", "")

        # Initialize Driver
        driver = InMemoryDriver(state_factory)
        driver = InMemoryDriver(run_id=run_id_, state_factory=state_factory)
```

Reply:
Do we still need the `run_id` arg after making the change you proposed?